### ISSUE TITLE
Compiler crash, wrong public function params?

### ISSUE THREAD

**Issue by Vince0789** [Zeex/pawn#664]
### Issue description:
Not sure how to describe this really, but at @Y-Less request I'm opening an issue here. Basically if you have the callbacks (see below) used in Pawn.CMD and you then include y_commands the compiler crashes.

### Minimal complete verifiable example (MCVE):
How to reproduce: 

* scaffold a new sampctl package with default settings
* require `sampctl/samp-stdlib` 
* require `pawn-lang/YSI-Includes:v5.05.0301`. 
* Super basic test script with the callbacks from Pawn.CMD:

```pawn
#include <a_samp>
#include <YSI_Visual\y_commands>

public OnPlayerCommandReceived(playerid, cmd[], params[], flags)
{
    return 1;
}

public OnPlayerCommandPerformed(playerid, cmd[], params[], result, flags)
{
    return 1;
}
```

### Workspace Information:

* Compiler version: 3.10.10
* Command line arguments provided (or sampctl version): sampctl 1.9.1
* Operating System: Windows 10 Pro, 20H2, build 19042.985

**Comment by killermvc** [comment]
After tinkering a bit with the code provided by Vince i was able to reduce it to this

```pawn
forward OnCallback(arr[]);
public OnCallback(array[]) {}
```

This seems to happen when the name of an array in the forward declaration isn't the same as in the public.
When telling sampctl to compile with compiler version `3.10.9` the compiler outputs the expected `(error) function heading differs from prototype` for both Vince's MCVE and mine.
The compiler also crashes when compiling with `-l`.

This was also reported in #637.

**Comment by Daniel-Cortez** [comment]
A duplicate of #600, #637. Already fixed in #561.
